Submitted By: Ryan Oliver <ryan dot oliver at pha dot com dot au>
Date: 2010-08-01
Initial Package Version: 5.8.8
Origin: Ryan Oliver
Upstream Status: Unknown
Description: This patch allows perl to be installed in /usr/lib32
             or /usr/lib64 instead of /usr/lib.

Rediffed: Against 5.10.0 by Joe Ciccone on 2008-08-30
          Against 5.10.1 By Jim Gifford on 2009-08-25
          Against 5.12.1 By Joe Ciccone on 2010-08-01

diff -Naur perl-5.12.1.orig/Configure perl-5.12.1/Configure
--- perl-5.12.1.orig/Configure	2010-05-13 18:01:07.000000000 -0400
+++ perl-5.12.1/Configure	2010-08-01 18:18:19.614326794 -0400
@@ -6254,6 +6254,8 @@
 : The default "style" setting is made in installstyle.U
 case "$installstyle" in
 *lib/perl5*) set dflt privlib lib/$package/$version ;;
+*lib32/perl5*) set dflt privlib lib32/$package/$version ;;
+*lib64/perl5*) set dflt privlib lib64/$package/$version ;;
 *)	 set dflt privlib lib/$version ;;
 esac
 eval $prefixit
@@ -6502,6 +6504,8 @@
 case "$sitelib" in
 '') case "$installstyle" in
 	*lib/perl5*) dflt=$siteprefix/lib/$package/site_$prog/$version ;;
+	*lib32/perl5*) dflt=$siteprefix/lib32/$package/site_$prog/$version ;;
+	*lib64/perl5*) dflt=$siteprefix/lib64/$package/site_$prog/$version ;;
 	*)	 dflt=$siteprefix/lib/site_$prog/$version ;;
 	esac
 	;;
@@ -7020,6 +7024,8 @@
 		prog=`echo $package | $sed 's/-*[0-9.]*$//'`
 		case "$installstyle" in
 		*lib/perl5*) dflt=$vendorprefix/lib/$package/vendor_$prog/$version ;;
+		*lib32/perl5*) dflt=$vendorprefix/lib32/$package/vendor_$prog/$version ;;
+		*lib64/perl5*) dflt=$vendorprefix/lib64/$package/vendor_$prog/$version ;;
 		*)	     dflt=$vendorprefix/lib/vendor_$prog/$version ;;
 		esac
 		;;
